File: c++/1393.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> x(40, 0);

void fibo() {
    x[0] = 1;
    x[1] = 2;
    for(int i = 2; i < 40; i++){
        x[i] = x[i - 1] + x[i - 2];
    }
}

int main() {
    int a;
    fibo();
    while(cin>>a && a)
        cout<<x[a - 1]<<endl;
}

File: python/1393.py
x=[1, 2]

def fibo():
    for i in range(2, 40):
        x.append(x[i - 1] + x[i - 2])

fibo()
while True:
    f = int(input())
    if f == 0:
        break
    print(x[f - 1])